<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PasswordGeneratorWidget</class>
 <widget class="QWidget" name="PasswordGeneratorWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>483</width>
    <height>205</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string/>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="labelNewPassword">
       <property name="text">
        <string>Password:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="sizeConstraint">
        <enum>QLayout::SetNoConstraint</enum>
       </property>
       <item>
        <widget class="QLineEdit" name="editNewPassword">
         <property name="maxLength">
          <number>999</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="togglePasswordButton">
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="labelLength">
       <property name="text">
        <string>Length:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <widget class="QSlider" name="sliderLength">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>128</number>
         </property>
         <property name="sliderPosition">
          <number>20</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::TicksBelow</enum>
         </property>
         <property name="tickInterval">
          <number>8</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinBoxLength">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>999</number>
         </property>
         <property name="value">
          <number>20</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="labelEntropy">
       <property name="text">
        <string>Entropy:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_5">
      <item>
        <widget class="QLabel" name="strengthLabel">
         <property name="minimumSize">
          <size>
           <width>70</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>strength</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QProgressBar" name="entropyProgressBar">
         <property name="maximum">
          <number>200</number>
         </property>
         <property name="value">
          <number>0</number>
         </property>
         <property name="invertedAppearance">
          <bool>false</bool>
         </property>
         <property name="textVisible">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="entropyLabel">
         <property name="minimumSize">
          <size>
           <width>70</width>
           <height>0</height>
          </size>
          </property>
          <property name="text">
            <string>entropy</string>
          </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>Character Types</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <widget class="QToolButton" name="checkBoxUpper">
            <property name="toolTip">
             <string>Upper Case Letters</string>
            </property>
            <property name="text">
             <string notr="true">A-Z</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">optionButtons</string>
            </attribute>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="checkBoxLower">
            <property name="toolTip">
             <string>Lower Case Letters</string>
            </property>
            <property name="text">
             <string notr="true">a-z</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">optionButtons</string>
            </attribute>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="checkBoxNumbers">
            <property name="toolTip">
             <string>Numbers</string>
            </property>
            <property name="text">
             <string notr="true">0-9</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">optionButtons</string>
            </attribute>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="checkBoxSpecialChars">
            <property name="toolTip">
             <string>Special Characters</string>
            </property>
            <property name="text">
             <string notr="true">/*_&amp; ...</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">optionButtons</string>
            </attribute>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QCheckBox" name="checkBoxExcludeAlike">
          <property name="text">
           <string>Exclude look-alike characters</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">optionButtons</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkBoxEnsureEvery">
          <property name="text">
           <string>Pick characters from every group</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">optionButtons</string>
          </attribute>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QPushButton" name="buttonGenerate">
         <property name="text">
          <string>Generate</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="buttonApply">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Accept</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>editNewPassword</tabstop>
  <tabstop>togglePasswordButton</tabstop>
  <tabstop>sliderLength</tabstop>
  <tabstop>spinBoxLength</tabstop>
  <tabstop>checkBoxUpper</tabstop>
  <tabstop>checkBoxLower</tabstop>
  <tabstop>checkBoxNumbers</tabstop>
  <tabstop>checkBoxSpecialChars</tabstop>
  <tabstop>checkBoxExcludeAlike</tabstop>
 </tabstops>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="optionButtons">
   <property name="exclusive">
    <bool>false</bool>
   </property>
  </buttongroup>
 </buttongroups>
</ui>
